repository/src/main/java/org/apache/atlas/tasks/TaskQueueWatcher.java

```diff
@@ -68,18 +72,13 @@ public void run() {
         if (LOG.isDebugEnabled()) {
             LOG.debug("TaskQueueWatcher: running {}:{}", Thread.currentThread().getName(), Thread.currentThread().getId());
         }
+        InterProcessMutex lock = null;
 
         while (shouldRun.get()) {
             try {
-                if (latch != null && latch.getCount() != 0) {
-                    LOG.info("TaskQueueWatcher: Waiting on Latch, current count: {}", latch.getCount());
-                    latch.await();
-                }
-
-                if (latch != null) {
-                    LOG.info("TaskQueueWatcher: Latch wait complete!!");
+                if (isActiveActiveHAEnabled) {
+                    lock = acquireDistributedLock();
```

Comment:
Asking out of curiosity what happens if the lock is acquired by another node, does code remain stuck here until other pod releases lock?

Reply:
Yes